app/controllers/katello/api/v2/upstream_subscriptions_controller.rb

```diff
@@ -22,9 +22,12 @@ class Api::V2::UpstreamSubscriptionsController < Api::V2::ApiController
     param :quantities_only, :bool, desc: N_("Only returns id and quantity fields")
     param :attachable, :bool, desc: N_("Return only subscriptions which can be attached to the upstream allocation")
     def index
-      pools = UpstreamPool.fetch_pools(upstream_pool_params.to_h)
+      index_params = upstream_pool_params
+      pools = UpstreamPool.fetch_pools(index_params)
+      page = index_params[:page] || 1
```

Comment:
Why not have this in `upstream_pool_params`?

Reply:
I wanted to set page=1 in katello's response when no page parameter is given since that appears to be how the rest of our API behaves, but since we could be removing 'page' from upstream_pool_params I did it outside of the method. also, I didn't want to put katello behavior in what is basically a candlepin method

Reply:
It just seemed similar to what’s in there already, setting defaults on unspecified params. I’m fine with the way it is.

Reply:
It has to be this way for the most part. If we want "full_result" we can't have `:page` coming out of `upstream_pool_params` or candlepin will not give the full result